src/librustdoc/config.rs

```diff
@@ -32,6 +33,8 @@ pub struct Options {
     pub input: PathBuf,
     /// The name of the crate being documented.
     pub crate_name: Option<String>,
+    /// Whether or not this is a proc-macro crate
+    pub proc_macro_crate: bool,
```

Comment:
```suggestion
    pub crate_types: Vec<CrateType>,
```
No need to go through two steps with a boolean in the middle if you can immediately obtain the end result in `fn from_matches`.

Reply:
I've not sure what you mean. This is the final step - all we care about is whether we were passed `CrateType::ProcMacro`, so we convert the `Vec<CrateType>` to a `bool` later in this file.

Reply:
I mean the only uses of `proc_macro_crate` are in `librustdoc/core.rs` and in `librustdoc/test.rs` where it's converted back into `Vec<CrateType>`.

Reply:
Those construct different `CrateType`s - one uses `CrateType::Rlib`, and the other uses `CrateType::Dylib`. Consumers of `Options` don't need to care which specific crate types were passed in on the command line.

Reply:
Ah, I see, I assumed they were the same.